near-sdk/src/json_types/public_key.rs

```diff
@@ -67,17 +80,19 @@ impl Base58PublicKey {
 
 impl From<Base58PublicKey> for Vec<u8> {
     fn from(v: Base58PublicKey) -> Vec<u8> {
-        v.0
+        let mut out = vec![v.curve as u8];
```

Comment:
I don't think casting the enum to u8 is safe for all platforms unless the enum is `repr(u8)`, or am I mistaken?

Probably best to have some conversion to u8

Reply:
hmm, the compiler doesn't complain about it though. And I can't seem to find anything on it being unsafe to cast. https://doc.rust-lang.org/nomicon/casts.html says `field-less enums as integer` are considered to be true casts.

Reply:
ah, that seems to be correct. Also seems the compiler is already having the repr being one byte so the explicit `repr(u8)` probably doesn't matter, although I'm not sure if it's best practice when casts like this are used.

Reply:
true, I always favor explicit over implicit. Marking it as `repr(u8)` shouldn't be an issue